### PR TITLE
aemu: restrict to supported platforms

### PR DESCRIPTION
**Pull request by NickCao**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/libraries/rutabaga_gfx/default.nix [hunk 1]
```diff
@@ -36,6 +36,6 @@ stdenv.mkDerivation (finalAttrs: {
     description = "cross-platform abstraction for GPU and display virtualization";
     license = licenses.bsd3;
     maintainers = with maintainers; [ qyliss ];
-    platforms = platforms.darwin ++ platforms.linux;
+    platforms = gfxstream.meta.platforms;
```
Comment (alyssais) [inline pkgs/development/libraries/rutabaga_gfx/default.nix:39]: Technically rutabaga doesn't require gfxstream (and therefore aemu), so I wonder whether we could omit that support on those platforms.  (Wayland forwarding should still work.)

**Comment by NickCao** [comment]
For some reason `rutabaga_gfx` is not properly producing binaries of the correct platform when cross compiling.

**Comment by NickCao** [comment]
> For some reason `rutabaga_gfx` is not properly producing binaries of the correct platform when cross compiling.

See also: https://github.com/NixOS/nixpkgs/issues/263019

File: pkgs/development/libraries/rutabaga_gfx/default.nix [hunk 2]
```diff
@@ -13,9 +19,17 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-0RJDKzeU7U6hc6CLKks8QcRs3fxN+/LYUbB0t6W790M=";
   };
 
+  patches = [
+    # Make gfxstream optional
+    # https://chromium-review.googlesource.com/c/crosvm/crosvm/+/4860836
+    (fetchpatch {
+      url = "https://github.com/google/crosvm/commit/c3ad0e43eb12cbf737a6049e0134d483e337363f.patch";
```
Comment (alyssais) [inline pkgs/development/libraries/rutabaga_gfx/default.nix:26]: Let's fetch this from upstream, not a mirror:

```suggestion
      url = "https://chromium.googlesource.com/crosvm/crosvm/+/c3ad0e43eb12cbf737a6049e0134d483e337363f%5E%21/?format=TEXT";
      decode = "base64 -d";
```

**Review by alyssais — APPROVED** [review]
diff lgtm

**Comment by alyssais** [comment]
> For some reason `rutabaga_gfx` is not properly producing binaries of the correct platform when cross compiling.

Fix: https://github.com/NixOS/nixpkgs/pull/310783

**Pull request merged**